src/System.Management.Automation/help/CommandHelpProvider.cs

```diff
@@ -511,16 +512,26 @@ private string GetHelpFile(string helpFile, CmdletInfo cmdletInfo)
                     // we have to search only in the application base for a mshsnapin...
                     // if you create an absolute path for helpfile, then MUIFileSearcher
                     // will look only in that path.
-                    helpFileToLoad = Path.Combine(mshSnapInInfo.ApplicationBase, helpFile);
+                    
+                    searchPaths.Add(mshSnapInInfo.ApplicationBase);
+                    searchPaths.Add(HelpUtils.GetUserHomeHelpSearchPath());
                 }
                 else if (cmdletInfo.Module != null && !string.IsNullOrEmpty(cmdletInfo.Module.Path))
-                {
-                    helpFileToLoad = Path.Combine(cmdletInfo.Module.ModuleBase, helpFile);
+                {                    
+                    searchPaths.Add(cmdletInfo.Module.ModuleBase);
+                    searchPaths.Add(HelpUtils.GetModuleBaseForUserHelp(cmdletInfo.Module.ModuleBase, cmdletInfo.Module.Name));
                 }
                 else
                 {
                     searchPaths.Add(GetDefaultShellSearchPath());
                     searchPaths.Add(GetCmdletAssemblyPath(cmdletInfo));
+
+                    // In case of Microsoft.PowerShell.Core module, cmdletInfo.ModuleName is null.
```

Comment:
I don't understand the comment here. For cmdlets from `Microsoft.PowerShell.Core`, `mshSnapInInfo` won't be `null` and thus will be covered by the first if block.

Reply:
Removed the comment. The code was refactored in to helper in HelpUtils.cs. The comment was left behind.